### PR TITLE
Make serialization and deserialization with additional validations cl…

### PR DESCRIPTION
**Pull request by NightMare489**
…oses #5

Added the following enhancements to the serialization and deserialization processes in serializers.py:

- **Name Validation**: Updated regular expression to allow Arabic characters in addition to English letters and spaces.
- **Field Length Checks**: Added checks to ensure the 'name' field does not exceed 50 characters and the 'msg' field does not exceed 500 characters.
- **Age Validation**: Age cannot be bigger than today
- **Default Values**: Ensured optional fields are handled correctly with default values.

These improvements should ensure that only valid data is serialized and deserialized, with clear error messages for any issues encountered.

### CODE REVIEW

File: back/app/serializers.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+class SerializationError(Exception):
+    pass
+
+class DeserializationError(Exception):
+    pass
+
+def serialize_data(data_model):
+    if not isinstance(data_model, Data):
+        raise SerializationError("Input is not a Data model instance.")
```
Comment (DevAbdoTolba) [inline back/app/serializers.py]: Better to raise a TypeError if the "type" did not match the intended type

**Review by DevAbdoTolba — CHANGES_REQUESTED** [review]
Looks great 🫡. A tiny change is required

**Review by NightMare489 — COMMENTED** [review]
Okay I have modifed it, Thank you for your feedback :)

**Review by DevAbdoTolba — APPROVED** [review]
LGTM 👍

**Pull request merged**